sphinx_immaterial/apidoc/python/parameter_objects.py

```diff
@@ -1,18 +1,42 @@
-from typing import Optional, cast, List, Dict, Sequence, Tuple, Any, Iterator
+from typing import (
+    Optional,
+    cast,
+    List,
+    Dict,
+    Sequence,
+    Tuple,
+    Any,
+    Iterator,
+    Literal,
+)
 
 import docutils.nodes
 from sphinx.domains.python import PyTypedField
 from sphinx.domains.python import PythonDomain
 from sphinx.domains.python import PyObject
+from sphinx.locale import _
 import sphinx.util.logging
+import sphinx.ext.autodoc
+import sphinx.util.typing
+import sphinx.ext.autodoc.typehints
+import sphinx.util.inspect
+import sphinx.ext.napoleon.docstring
+from sphinx.ext.napoleon.docstring import GoogleDocstring
+
 
 from . import annotation_style
 from .. import object_description_options
 from ... import sphinx_utils
 
-
 logger = sphinx.util.logging.getLogger(__name__)
 
+if sphinx.version_info >= (7, 1):
+    from sphinx.addnodes import desc_type_parameter
+else:
+    # Fake class definition that will always fail instanceof checks.
```

Comment:
```suggestion
    # Fake class definition that will always fail isinstance() checks.
```
this is in python after all 😜